chain/store/store.go

```diff
@@ -210,6 +211,38 @@ func (cs *ChainStore) SubscribeHeadChanges(f func(rev, app []*types.TipSet) erro
 	cs.headChangeNotifs = append(cs.headChangeNotifs, f)
 }
 
+func (cs *ChainStore) IsTipSetValidated(ctx context.Context, ts *types.TipSet) (bool, error) {
+	for _, b := range ts.Blocks() {
+		isValidated, err := cs.IsBlockValidated(ctx, b)
+
+		if err != nil {
+			return false, err
+		}
+
+		if !isValidated {
+			return false, nil
+		}
+	}
+
+	return true, nil
+}
+
+func (cs *ChainStore) IsBlockValidated(ctx context.Context, blk *types.BlockHeader) (bool, error) {
+	key := tipsetBlockValidationCacheKeyPrefix.Instance(blk.Cid().String())
+
+	return cs.ds.Has(key)
+}
+
+func (cs *ChainStore) MarkBlockAsValidated(ctx context.Context, blk *types.BlockHeader) error {
```

Comment:
Can all of the methods added to this file take block CIDs / TipSetKeys as parameters, instead of the BlockHeader / TipSet?

Reply:
fixed